client/landing/stepper/declarative-flow/flows/new-hosted-site-flow/new-hosted-site-flow.ts

```diff
@@ -63,47 +99,59 @@ const hosting: Flow = {
 		);
 
 		const query = useQuery();
-		const queryParams = Object.fromEntries( query );
-		const plan = queryParams.plan;
+
+		const utmSource = query.get( 'utm_source' );
+		const studioSiteId = query.get( 'studioSiteId' );
+
 		const flowName = this.name;
-		const showDomainStep = useShowDomainStep();
+		const showDomainStep = query.has( 'showDomainStep' );
 		const isWooPartner = useIsValidWooPartner();
 
 		const goBack = () => {
-			if ( _currentStepSlug === 'plans' ) {
+			if ( _currentStepSlug === STEPS.UNIFIED_PLANS.slug ) {
 				if ( showDomainStep ) {
-					return navigate( 'domains' );
+					return navigate( STEPS.UNIFIED_DOMAINS.slug );
 				}
 				return window.location.assign( '/sites?hosting-flow=true' );
 			}
-			if ( _currentStepSlug === 'trialAcknowledge' ) {
-				navigate( 'plans' );
+			if ( _currentStepSlug === STEPS.TRIAL_ACKNOWLEDGE.slug ) {
+				navigate( STEPS.UNIFIED_PLANS.slug );
 			}
 		};
 
+		const flowSteps = this.getSteps!();
+
 		const submit = ( providedDependencies: ProvidedDependencies = {} ) => {
 			if ( providedDependencies.siteId ) {
 				setSignupCompleteSiteID( providedDependencies.siteId );
 			}
 
 			switch ( _currentStepSlug ) {
-				case 'domains': {
+				case STEPS.UNIFIED_DOMAINS.slug: {
```

Comment:
This is a great change but should be tested well. The unified domain step has weird issues when the flow doesn't [opt-in](https://github.com/Automattic/wp-calypso/blob/7aa76feb649585d92701461845ac93d925c6b3d7/client/signup/steps/domains/utils.js#L36) to multi-domain selection. Frankly, I'd activate multi domain selection for every flow.

Reply:
Absolutely. You're right that most flows don't, but most (90%+) users are offered the multi-domain options because it's the one running in the onboarding flow. This means the unified-domains step is not battle tested when it comes to single domain selection.